### ISSUE TITLE
Can't update a Github release using the set_github_release action

### ISSUE THREAD

**Issue by Goles** [fastlane/fastlane#4704]
### Issue Description

I'm trying to update an already existing Github release for a certain `release-tag`. The aforementioned release has been marked as a `prerelease` in Github. What I want, is to update the `prerelease` to `release`.

Problem is that apparently the `set_github_release` action wasn't designed to allow updating of an already created Github Release. So I just get an `Release on tag test-tag already exists!` error message.

```
fastlane release version:test-tag
[10:40:16]: --------------------------------
[10:40:16]: --- Step: get_github_release ---
[10:40:16]: --------------------------------
[10:40:16]: Getting release on GitHub (https://api.github.com/Vungle/ios-sdk-for-real: test-tag)
[10:40:17]: Version is already live on GitHub.com 🚁
[10:40:17]: --------------------------------
[10:40:17]: --- Step: set_github_release ---
[10:40:17]: --------------------------------
[10:40:17]: Creating release of Vungle/ios-sdk-for-real on tag "test-tag" with name "Release version test-tag".
[10:40:17]: {"message":"Invalid request.\n\nFor 'properties/body', nil is not a string.","documentation_url":"https://developer.github.com/v3"}
[10:40:17]: Release on tag test-tag already exists!
```

### Configuration Files

Fastfile:
```ruby
  desc "Upgrades a Github prelease to release"
  lane :release do |options|

    # current version that it's being released
    version = options[:version]

    if not version
        abort "Error: no release version specified"
    end

    # A prerelease must exist for a release to exist
    release = get_github_release(url: "MyRepo/MyProject",
                                                    api_token: ENV['GITHUB_TOKEN'],
                                                    version: version)


    if not ENV['GITHUB_TOKEN']
      abort "Error: $GITHUB_TOKEN not set"
    end

    set_github_release(
      repository_name: "MyRepo/MyProject",
      api_token: ENV['GITHUB_TOKEN'],
      is_prerelease: false, 					# Note that I'm setting prerelease to false
      tag_name: version,
      name: "Release version " + version,
    )
  end
```

### Environment

`fastlane 1.89.0`
`ruby version 2.0.0-p643`
using `rbenv`

**Comment by robbiet480** [comment]
I'm having the same issue, except in my case I don't have any existing releases and am just trying to create a new one for the first time.

**Comment by yanndupuy** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  dupuyyann@gmail.com
    Domain zyapps.com has exceeded the max emails per hour (408/400 (102%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o6.sgmail.github.com ([192.254.113.101]:34946)
	by serveur.ex2-p10.com with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.87)
	(envelope-from <bounces+848413-8f06-yann.dupuy=zyapps.com@sgmail.github.com>)
	id 1b8IZ1-002DW4-PY
	for yann.dupuy@zyapps.com; Thu, 02 Jun 2016 04:41:02 +0200
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=oroc2/ONHgV6XmSd+6Pa1aedK4E=; b=MTjSS1zShn1nwzI4
	iPQXEwhIV98CRzVE9HcI/m7pkUwGtDEORVQVZRxEd+6+DLzesHZuS0eKs//dJaOD
	DoQBt5Ex7ppYpNjNgZj4NiNkNvhl5DpBYOmmHeKbEUprY5g07HUDubG9bOpQk7CQ
	Rw6Mq0F7uf+fQE1uoUg2XLmHxik=
Received: by filter0428p1mdw1.sendgrid.net with SMTP id filter0428p1mdw1.5810.574F9C8314
        2016-06-02 02:40:03.17808975 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd0002p1iad1.sendgrid.net (SG) with ESMTP id FbgnM7_LSG2Ju7Tgtypstw
	for <yann.dupuy@zyapps.com>; Thu, 02 Jun 2016 02:40:03.301 +0000 (UTC)
Date: Wed, 01 Jun 2016 19:40:03 -0700

**Comment by vpolouchkine** [comment]
@robbiet480 Can you please provide more details? The action does not work out of the box?

**Comment by robbiet480** [comment]
@vpolouchkine Correct. Here's the exact action I was using:

```ruby
set_github_release(
      repository_name: "home-assistant/home-assistant-iOS",
      api_token: ENV["GITHUB_API_TOKEN"],
      name: complete_version_number,
      tag_name: tag_name("testflight", version_number, build_number),
      description: change_log_since_last_tag,
      is_prerelease: false
    )
```

I got it from [the Fabric example](https://github.com/fastlane/examples/blob/master/Fabric/Fastfile#L79-L87).

I know the Github token is valid, because I even regenerated it and still had the same error as @Goles (`{"message":"Invalid request.\n\nFor 'properties/body', nil is not a string.","documentation_url":"https://developer.github.com/v3"}`).

`fastlane 1.93.1`
`ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]`
Also using `rbenv`.

**Comment by vpolouchkine** [comment]
Can you try setting the `commitish` argument?

**Comment by vpolouchkine** [comment]
It looks like there hasn’t been any recent activity on this issue in some time, so I’m closing it. If this is still a problem, feel free to re-open a new issue and I’d be happy to help :+1:

**Comment by jmbeck** [comment]
@robbiet480 - I just encountered the same issue, the issue was that the `description` property was empty for me, meaning your `change_log_since_last_tag` must be nil.

I changed mine to `description: change_log_since_last_tag || "Release commit"`, so it still works when `change_log_since_last_tag` is nil.

It must only cause issues when attempting to use fastlane many times in succession (such as when testing your Fastfile) without any commits of any substance in between releases.
